homeassistant/components/advantage_air/climate.py

```diff
@@ -101,17 +103,42 @@ def __init__(self, instance: AdvantageAirData, ac_key: str) -> None:
             HVACMode.FAN_ONLY,
             HVACMode.DRY,
         ]
-        # Set supported features and HVAC modes based on current operating mode
-        if self._ac.get(ADVANTAGE_AIR_MYAUTO_ENABLED):
-            # MyAuto
-            self._attr_supported_features |= (
-                ClimateEntityFeature.TARGET_TEMPERATURE
+
+        # Add "MyTemp" preset if available
+        if ADVANTAGE_AIR_MYTEMP_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYTEMP]
+            self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
+
+        # Add "MyAuto" preset if available
+        if ADVANTAGE_AIR_MYAUTO_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYAUTO]
+            self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
```

Comment:
`_attr_preset_modes` and `_attr_supported_features` shouldn't be set anymore because you are overriding the shorthand attributes by defining `preset_mode` and `supported_features`.  It needs to be one or the other

Its a bit faster to use `_attr_preset_modes` and `_attr_supported_features` and update them when they change because they are `@cached_property`

Reply:
Ill refactor to use `_handle_coordinator_update `

Reply:
@bdraco I have refactored to use _handle_coordinator_update  rather than `preset_mode` and `supported_features`. In a future PR I will also refactor many of the other properties to use the same function.